### DISCUSSION TITLE
how can I get the isolated nodes in a graph?

### DISCUSSION THREAD

**Discussion by BodyCSoulN** [pyg-team/pytorch_geometric#4130] [Q&A] [answered]
Hi, thanks for your great job.
There is method called `has_isolated_nodes()`.
I want to know how can I get these isolated nodes?

Because I have a graph stored in a neo4j database.

I can't find a isolated node in neo4j.
But when I convert this graph to pyg data, `has_isolated_nodes()` returns True.

I don't know what went wrong？
Can you help me.

**Comment by rusty1s** [comment, marked as answer]
How about
```python
unique = data.edge_index.view(-1).unique()

mask = torch.ones(data.num_nodes, dtype=torch.bool)
mask[unique] = False

print(mask.nonzero())  # The indices of all isolated nodes.
```
